core/internal/cltest/cltest.go

```diff
@@ -1463,3 +1467,10 @@ func MustDefaultKey(t *testing.T, s *strpkg.Store) models.Key {
 	require.NoError(t, err)
 	return k
 }
+
+func RandomiseNonce(t *testing.T, s *strpkg.Store) {
```

Comment:
How does this prevent deadlocks?

Reply:
It prevents multiple transactions from attempting to insert eth_txes with the same nonce (locking on unique index creates problems)

Reply:
Shouldn't the databases be isolated for each test and in transactions?